crates/turbopack-ecmascript/src/parse/top_level_await.rs

```diff
@@ -0,0 +1,17 @@
+use swc_core::ecma::{ast::*, visit::Visit};
+
+#[derive(Default)]
+pub(super) struct TopLevelAwaitVisitor {
+    pub(super) has_top_level_await: bool,
+}
+
+impl Visit for TopLevelAwaitVisitor {
+    fn visit_await_expr(&mut self, _: &AwaitExpr) {
+        self.has_top_level_await = true;
+    }
+
+    // prevent non top level items from visiting their children
+    fn visit_arrow_expr(&mut self, _: &ArrowExpr) {}
+    fn visit_class(&mut self, _: &Class) {}
+    fn visit_function(&mut self, _: &Function) {}
```

Comment:
I don't think `function` is an alias the way Babel uses it. Instead, we'd need more like  https://github.com/swc-project/swc/blob/main/crates/swc_ecma_transforms_compat/src/es2022/class_properties/used_name.rs#L21-L31

Reply:
Thought class would cover everything, but I forgot about objects and we need some parts of the class now anyways

Reply:
yeah I kinda forgot about object getters and setters